src/facade/memcache_parser.cc

```diff
@@ -103,52 +111,28 @@ MP::Result ParseValueless(const std::string_view* tokens, unsigned num_tokens, M
 
 auto MP::Parse(string_view str, uint32_t* consumed, Command* cmd) -> Result {
   cmd->no_reply = false;  // re-initialize
-  auto pos = str.find('\n');
+  auto pos = str.find("\r\n");
   *consumed = 0;
   if (pos == string_view::npos) {
-    // TODO: it's over simplified since we may process GET/GAT command that is not limited to
-    // 300 characters.
-    return str.size() > 300 ? PARSE_ERROR : INPUT_PENDING;
+    return INPUT_PENDING;
   }
 
   if (pos == 0) {
     return PARSE_ERROR;
   }
-  *consumed = pos + 1;
+  *consumed = pos + 2;
+
+  std::string_view tokens_expression = str.substr(0, pos);
 
   // cas <key> <flags> <exptime> <bytes> <cas unique> [noreply]\r\n
   // get <key>*\r\n
-  string_view tokens[8];
-  unsigned num_tokens = 0;
-  uint32_t cur = 0;
-
-  while (cur < pos && str[cur] == ' ')
-    ++cur;
-
-  uint32_t s = cur;
-  for (; cur <= pos; ++cur) {
-    if (absl::ascii_isspace(str[cur])) {
-      if (cur != s) {
-        tokens[num_tokens++] = str.substr(s, cur - s);
-        if (num_tokens == ABSL_ARRAYSIZE(tokens)) {
-          ++cur;
-          s = cur;
-          break;
-        }
-      }
-      s = cur + 1;
-    }
-  }
+  absl::InlinedVector<std::string_view, 32> tokens = absl::StrSplit(tokens_expression, ' ');
```

Comment:
absl::StrSplit will always allocate a vector.

Reply:
you mean a `std::vector` ? That means that an `InlinedVector` is implicitly converted to `std::vector` ?

Reply:
My mistake, it's not.